### ISSUE TITLE
Document config.h options and effects

### ISSUE THREAD

**Issue by haasn** [qmk/qmk_firmware#1313]
As far as I can tell, there's no place to get information about available `config.h` options and their effects other than reading the source code.

For example, I recently figured out that I can seemingly use config.h changes to solve **two** of the annoying bugs I had with qmk_firmware (`#define PREVENT_STUCK_MODIFIERS` and `#undef IGNORE_MOD_TAP_INTERRUPT`, respectively); and I only found out about these while reading other people's various `config.h` files.

It's annoying that I can't really find out more information about what they actually do, other than a vague guess based on the title. If there is documented at all in the repo or the website, I cannot find it.

I would probably suggest adding a commented-out version of all available options + documentation to the “config.h” template.

**Comment by jackhumbert** [comment]
PREVENT_STUCK_MODIFIERS is described [here](https://github.com/qmk/qmk_firmware_docs/blob/master/Keymap-examples.md) - this isn't on the docs site yet because that page isn't linked in the summary. We're still doing a lot of organising there :)

**Comment by jayliu50** [comment]
A small plug for a request to document the ability to redefine magic modifier for magic key command, as accidentally activating it has tripped up some of us, myself included  :)  (#1233) 

```c
/* key combination for magic key command */
#define IS_COMMAND() ( \
    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
)
```
